iceoryx_posh/include/iceoryx_posh/internal/mepoo/shm_safe_unmanaged_chunk.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_MEPOO_SHM_SAFE_UNMANAGED_CHUNK_HPP
+#define IOX_POSH_MEPOO_SHM_SAFE_UNMANAGED_CHUNK_HPP
+
+#include "iceoryx_posh/internal/mepoo/shared_chunk.hpp"
+#include "iceoryx_utils/internal/relocatable_pointer/relative_pointer_data.hpp"
+
+namespace iox
+{
+namespace mepoo
+{
+class ShmSafeUnmanagedChunk
+{
+  public:
+    ShmSafeUnmanagedChunk() noexcept = default;
+
+    /// @brief takes a SharedChunk without decrementing the chunk reference counter
+    ShmSafeUnmanagedChunk(SharedChunk chunk) noexcept;
+
+    /// @brief Creates a SharedChunk without incrementing the chunk reference counter and invalidates itself
+    SharedChunk releaseToSharedChunk() noexcept;
+
+    /// @brief Creates a SharedChunk with incrementing the chunk reference counter and does not invalidate itself
+    SharedChunk duplicateToSharedChunk() noexcept;
```

Comment:
well, it increments the refcount and has to be seen in combination with `releaseToSharedChunk`

Reply:
create a new shared chunk without releasing ownership. maybe `copyToSharedChunk` or we use the conversion operator `operator SharedChunk()`. What do you think is more understandable.

Reply:
copy is ambiguous since this is unmanaged a plain copy would essentially be the same as `releaseToSharedChunk` but with leaving a a dangling pointer behind. I'm open to change it, but it should clearly show that the ref count is incremented

Reply:
how about `cloneToSharedChunk`?